graphics/blit/blit.cpp

```diff
@@ -127,59 +157,26 @@ inline void crossBlitLogic(byte *dst, const byte *src, const uint w, const uint
 	}
 }
 
-template<typename DstColor, bool backward, bool hasKey>
+template<typename DstColor, int DstSize, bool backward, bool hasKey>
 inline void crossBlitLogic1BppSource(byte *dst, const byte *src, const uint w, const uint h,
 									 const uint srcDelta, const uint dstDelta, const uint32 *map, const uint32 key) {
 	for (uint y = 0; y < h; ++y) {
 		for (uint x = 0; x < w; ++x) {
 			const byte color = *src;
 			if (!hasKey || color != key) {
-				*(DstColor *)dst = map[color];
+				if (DstSize == sizeof(DstColor)) {
+					*(DstColor *)dst = map[color];
+				} else {
+					WRITE_UINT24(dst, map[color]);
```

Comment:
Considering this is specific to DstSize == 3, should the check conditions reflect that? Do we anticipate the other checks handling non-3 byte scenarios?